### ISSUE TITLE
Incompatible types when using esm

### ISSUE THREAD

**Issue by viceice** [steveukx/git-js#704]
Currently the exposed types are not usable, when compiling typescript code as esm.

When importing this lib from esm, i need to write `import * as simpleGit from 'simple-git';`.

This will cause `simpleGit.default` to be the factory, which works.
But the other enum types are not accessable as `simpleGit.ResetMode`, instead they are exported via `simpleGit.default.ResetMode`.
Trying to use those will cause compilations issues.

https://github.com/renovatebot/renovate/pull/13184/files#diff-3c967a96825db779ae3fa03f549890aa993ea47b1b3326e800e7a582f1b5b351R43

**Comment by viceice** [comment]
maybe this

https://github.com/steveukx/git-js/blob/a569868d800a0d872e8fb1534bb0dceccff47a4f/typings/index.d.ts#L1-L11

 needs something like

```ts
import { SimpleGitFactory } from './simple-git';

declare namespace SimpleGitFactory {
  export * from './simple-git';
  export * from './errors';
  export * from './response';
  export * from './types';
}

export declare const gitP: SimpleGitFactory;

declare const simpleGit: SimpleGitFactory;
export = simpleGit;
```

**Comment by steveukx** [comment]
Hi, thank you for pointing this out.

Until I have a fix in the library, you can change the import to be:

```
import simpleGit from 'simple-git';
const {CleanOptions} = simpleGit;
```

**Comment by steveukx** [comment]
The next release will contain #708 which enables the ES module imports in the same format as they would appear in TypeScript, ie:

```
import simpleGit, { ResetMode } from 'simple-git';
```

You can follow the release in #709

**Comment by steveukx** [comment]
Hi, the latest version of `simple-git` is now available on npm, published with a major version bump as the packaging has changed to allow for both importing esm and requiring commonjs.

There are very few usage changes, so there should be no code change required in order to upgrade from v2.x to v3.x (unless you were using deprecated APIs that had been logging warnings to the console), or were importing from a path other than just `simple-git`.

There are notes for the release available in https://github.com/steveukx/git-js/tree/repo-v3.0.3#release-history and examples for ways to import the library in https://github.com/steveukx/git-js/tree/repo-v3.0.3#usage

Please do raise a new issue if you are having any problems using the new version, thanks.

**Comment by viceice** [comment]
Thanks, Will try in a few days, as it's currently blocked by renovate stability days. 😅
